app/controllers/ephemera_folders_controller.rb

```diff
@@ -88,13 +92,40 @@ def load_fields
     end
   end
 
+  # provides instance variables needed to populate collection and selected for append_id
+  # @see app/views/ephemera_folders/edit_fields/_append_id.html.erb
+  def load_boxes
+    instance_variable_set("@available_boxes", available_boxes)
+    instance_variable_set("@selected_box", ephemera_box&.id.to_s)
+    instance_variable_set("@add_to_box", ephemera_box.nil?)
+  end
+
+  # returns decorators
+  def available_boxes
+    if ephemera_project
+      ephemera_project.boxes
+    else
+      []
+    end.unshift(nil_box)
+  end
+
+  def ephemera_box
+    parent_resource.is_a?(EphemeraBox) ? parent_resource : nil
+  end
+
+  def nil_box
+    Struct.new(:title, :id).new('', '')
```

Comment:
What's the nil_box get us that the empty array doesn't?

Reply:
I actually don't think I tested the empty array case. When you have a non-empty array you need a nil option to keep the form from filling in the first value of the array, and allowing the user to not fill in the field if they don't want to.